app/controllers/v1/forum/threads_controller.rb

```diff
@@ -1,4 +1,14 @@
 module V1::Forum
   class ThreadsController < V1::ApplicationController
+    before_action :doorkeeper_authorize!
+    before_action :set_model, only: %i[mark_read]
+
+    def mark_read
+      last = @model.posts.last
+      thread = Forum::ReadThread.where(thread: @model, user: current_user).first_or_create! do |t|
+        t.post = last
+      end
+      thread.update(post: last)
```

Comment:
Isn't this duplicate with the above do block? I think we can remove the do block

Reply:
```
thread = Forum::ReadThread.first_or_create(user: User.first, thread: Forum::Thread.first)
thread.post = @model.posts.last
thread.save
```

Reply:
dit is nu aangepast